### PR TITLE
Patch openthread

### LINKED ISSUE #4: Get openthread configurations via backend of webui
Make request for getting the openthread configurations via backend of the webui instead of multicastDNS

### PR DESCRIPTION
**Pull request by lcc**
Get openthread configurations via backend of webui.

Closes #4.

### CODE REVIEW

File: app/src/main/java/br/org/cesar/knot_setup_app/activity/login/LoginPresenter.java [hunk 1]
```diff
@@ -51,14 +52,63 @@ public void doLogin(String password){
         this::onErrorHandler);
     }
 
+    private void getOpenthreadConfig(){
+        String bearer,ip;
+        bearer = dataManager.getInstance()
+                .getPersistentPreference()
+                .getSharedPreferenceString(context,"token");
+
+        ip = dataManager.getInstance()
```
Comment (jcorde) [inline app/src/main/java/br/org/cesar/knot_setup_app/activity/login/LoginPresenter.java]: Why this is not an item of `LoginPresenter`?

File: app/src/main/java/br/org/cesar/knot_setup_app/activity/login/LoginPresenter.java [hunk 2]
```diff
@@ -51,14 +52,63 @@ public void doLogin(String password){
...
+                .getPreference()
+                .getSharedPreferenceString(context,"ip");
+
+        this.port = dataManager.getInstance()
```
Comment (jcorde) [inline app/src/main/java/br/org/cesar/knot_setup_app/activity/login/LoginPresenter.java]: Why getting the port again?
You just got it at `LoginPresenter(ViewModel viewModel, Context context)`

File: app/src/main/java/br/org/cesar/knot_setup_app/activity/login/LoginPresenter.java [hunk 3]
```diff
@@ -51,14 +52,63 @@ public void doLogin(String password){
...
+
     private void loginSucceeded(User user) {
         dataManager.getInstance().getPersistentPreference()
                 .setSharedPreferenceString(context,"email",email);
 
         dataManager.getInstance().getPersistentPreference()
                 .setSharedPreferenceString(context,"token","Bearer " + user.getToken());
 
-        mViewModel.callbackOnLogin();
+        getOpenthreadConfig();
```
Comment (jcorde) [inline app/src/main/java/br/org/cesar/knot_setup_app/activity/login/LoginPresenter.java:105]: Are you using this anywhere else?
If not, please, remove it

**Review by jcorde — CHANGES_REQUESTED** [review]

File: app/src/main/java/br/org/cesar/knot_setup_app/activity/login/LoginPresenter.java (hunk 1 above)
Comment (lcc) [reply app/src/main/java/br/org/cesar/knot_setup_app/activity/login/LoginPresenter.java]: Done

File: app/src/main/java/br/org/cesar/knot_setup_app/activity/login/LoginPresenter.java (hunk 2 above)
Comment (lcc) [reply app/src/main/java/br/org/cesar/knot_setup_app/activity/login/LoginPresenter.java]: Done

File: app/src/main/java/br/org/cesar/knot_setup_app/activity/login/LoginPresenter.java (hunk 3 above)
Comment (lcc) [reply app/src/main/java/br/org/cesar/knot_setup_app/activity/login/LoginPresenter.java:105]: Will do when getting rid of leftovers.

Comment (jcorde) [reply app/src/main/java/br/org/cesar/knot_setup_app/activity/login/LoginPresenter.java:105]: This patch is supposed to be self-contained. Please, remove it or refer to a PR that removes it.

**Pull request merged**